src/test/java/org/xstefank/check/LatestCommitCheckTest.java

```diff
@@ -0,0 +1,80 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.model.yaml.FormatMatcher;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import static org.xstefank.check.AuxiliaryTestMethods.readJsonNode;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(GitHubAPI.class)
+public class LatestCommitCheckTest {
+
+    private FormatMatcher testCommitFormatMatcher = new FormatMatcher();
+    private LatestCommitCheck testLatestCommitCheck;
+    private static JsonNode testPullRequestPayload;
+    private static JsonNode testCommitsPayload;
+    private static final String stubMethodName = "getJsonWithCommits";
+    private static final String suppressMethodName = "readToken";
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        testPullRequestPayload = readJsonNode("testPullRequestPayload.json");
+        testCommitsPayload = readJsonNode("testCommitsPayload.json");
+    }
+
+    @Before
+    public void setUp() {
+        suppress(method(GitHubAPI.class, suppressMethodName));
+        PowerMockito.stub(PowerMockito.method(GitHubAPI.class, stubMethodName, JsonNode.class)).toReturn(testCommitsPayload);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullCommitParameter() throws IllegalArgumentException {
+        new LatestCommitCheck(null);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullCommitPatternParameter() throws IllegalArgumentException {
+        testCommitFormatMatcher.setPattern(null);
+        new LatestCommitCheck(testCommitFormatMatcher);
+    }
+
+    @Test
+    public void testCheckSimpleRegexMatch() {
+        testCommitFormatMatcher.setPattern(Pattern.compile("test commit"));
+        testLatestCommitCheck = new LatestCommitCheck(testCommitFormatMatcher);
+
+        Assert.assertNull("Cannot match valid regex", testLatestCommitCheck.check(testPullRequestPayload));
+    }
+
+    @Test
+    public void testCheckSimpleRegexNonMatchReturnExpectedMessage() {
+        testCommitFormatMatcher.setPattern(Pattern.compile("can't.*match.*this"));
+        testCommitFormatMatcher.setMessage("This is testCommitFormatMatcher message.");
+        testLatestCommitCheck = new LatestCommitCheck(testCommitFormatMatcher);
+
+        Assert.assertNotNull("Matched invalid regex", testLatestCommitCheck.check(testPullRequestPayload));
+        Assert.assertEquals("Not returned expected message", "This is testCommitFormatMatcher message.", testLatestCommitCheck.check(testPullRequestPayload));
```

Comment:
this should be a full sentence, it won't make sense to the caller when this test fails - you can write "Unexpected message returned" or "method call did not return..."